modules/createHashHistory.js

```diff
@@ -83,7 +87,7 @@ const createHashHistory = (options = {}) => {
   }
 
   const createHref = (path) =>
-    '#' + history.createHref(path)
+    '#' + transformPath(history.createHref(path), true)
```

Comment:
Well, if we're going to use a `transformPath` function, can we at least make the default case use it as well? i.e. the default `transformPath` takes care of prepending the `#`.